fix(eslint): lint cypress files

jurajmajerik opened the pull request:
## Changes
Context: https://posthog.slack.com/archives/C0113360FFV/p1720447504686709

We want to include cypress files in the pre-commit check to enforce this rule: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-focused-tests.md

The `cypress` folder was previously part of `ignorePatterns`, not sure why, I guess @thmsobrmlr knows?

jurajmajerik committed:
init

thmsobrmlr commented:
> The `cypress` folder was previously part of `ignorePatterns`, not sure why, I guess @thmsobrmlr knows?

I think this was because eslint complains about unknown `expect` statements in the `*.cy.ts` files. You can see those by running `pnpm eslint cypress`. It would be great to get linting working here though, as running that statement actually shows a couple of warning like unused imports.

> We want to include cypress files in the pre-commit check to enforce this rule: [jest-community/eslint-plugin-jest@`main`/docs/rules/no-focused-tests.md](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-focused-tests.md?rgh-link-date=2024-07-09T07%3A17%3A16Z)

To do so, you'll need to adapt the `eslint` command used in CI as well & clean up the current problems.

jurajmajerik committed:
adjust overrides + package.json

jurajmajerik committed:
fix issues

jurajmajerik commented:
thanks @thmsobrmlr!

I have:
- applied relevant overrides in `.eslintrc.js` to the cypress files
- adjusted the CI command in `package.json` to also apply to the `cypress folder`
- fixed all eslint errors in the `cypress` folder

thmsobrmlr reviewed (approved):
Amazing, thank you!

jurajmajerik committed:
fix broken survey tests
